test/parallel/test-dgram-multicast-set-interface-lo.js

```diff
@@ -0,0 +1,275 @@
+        if (worker.isDone || code == 0) {
+          return;
+        }
+
+        dead += 1;
+        console.error('[PARENT] Worker %d died. %d dead of %d',
+                      worker.pid,
+                      dead,
+                      listeners);
+
+        if (dead === listeners) {
+          console.error('[PARENT] All workers have died.');
+          console.error('[PARENT] Fail');
+
+          killChildren(workers);
+
+          process.exit(1);
+        }
+      });
+
+      worker.on('message', function(msg) {
+        if (msg.listening) {
+          listening += 1;
+
+          if (listening === listeners) {
+            //all child process are listening, so start sending
+            sendSocket.sendNext();
+          }
+        } else if (msg.message) {
+          worker.messagesReceived.push(msg.message);
+
+          if (worker.messagesReceived.length === worker.messagesNeeded.length) {
+            done += 1;
+            worker.isDone = true;
+            console.error('[PARENT] %d received %d messages total.',
+                          worker.pid,
+                          worker.messagesReceived.length);
+          }
+
+          if (done === listeners) {
+            console.error('[PARENT] All workers have received the ' +
+                          'required number of ' +
+                          'messages. Will now compare.');
+
+            Object.keys(workers).forEach(function(pid) {
+              var worker = workers[pid];
+
+              var count = 0;
+
+              worker.messagesReceived.forEach(function(buf) {
+                for (var i = 0; i < worker.messagesNeeded.length; ++i) {
+                  if (buf.toString() === worker.messagesNeeded[i]) {
+                    count++;
+                    break;
+                  }
+                }
+              });
+
+              console.error('[PARENT] %d received %d matching messages.',
+                            worker.pid,
+                            count);
+
+              assert.equal(count, worker.messagesNeeded.length,
+                           'A worker received an invalid multicast message');
+            });
+
+            clearTimeout(timer);
+            console.error('[PARENT] Success');
+            killChildren(workers);
+          }
+        }
+      });
+    })(x);
+  }
+
+  var sendSocket = dgram.createSocket({
+    type: UDP[FAM],
+    reuseAddr: true
+  });
+
+  // don't bind the address explicitly for sending
+  sendSocket.bind(common.PORT, ANY[FAM]);
+  sendSocket.on('listening', function() {
+    // but explicitly set the outgoing interface
+    sendSocket.setMulticastInterface(interfaceAddress);
+    console.error(`outgoing iface ${interfaceAddress}`);
+  });
+
+  sendSocket.on('close', function() {
+    console.error('[PARENT] sendSocket closed');
+  });
+
+  sendSocket.sendNext = function() {
+    const msg = messages[i++];
+
+    if (!msg) {
+      try { sendSocket.close(); } catch (e) {}
```

Comment:
I'll remove this since I think it is worth seeing anything from the sender given the nature of this test. I think the reason it was there is since the parent is both the sender and responsible for cleanup, so it is a little more likely a failure in send side will bleed into subsequent tests by leaving orphans.